### PR TITLE
CORE-19752: Hide root cause message and only for exceptional cases

### PR DESCRIPTION
**Pull request by jennyang-r3**
Redact root exception message to a generic message, handle only exceptional cases

### CODE REVIEW

**Comment by jennyang-r3** [comment]
build e2e

File: notary-plugins/notary-plugin-non-validating/notary-plugin-non-validating-server/src/main/kotlin/com/r3/corda/notary/plugin/nonvalidating/server/NonValidatingNotaryServerFlowImpl.kt [hunk 1]
```diff
@@ -114,11 +116,18 @@ class NonValidatingNotaryServerFlowImpl() : ResponderFlow {
             session.send(uniquenessResult.toNotarizationResponse(txDetails.id, signature))
         } catch (e: Exception) {
             logger.warn("Error while processing request from client. Cause: $e ${e.stackTraceToString()}")
+            val genericMessage = "Error while processing request from client. "
+            val additionalMessage =
+                when (e) {
+                    is NotaryExceptionInputStateConflict -> "Cause: ${e.message}"
+                    is NotaryExceptionReferenceStateConflict -> "Cause: ${e.message}"
```
Comment (blsemo) [inline notary-plugins/notary-plugin-non-validating/notary-plugin-non-validating-server/src/main/kotlin/com/r3/corda/notary/plugin/nonvalidating/server/NonValidatingNotaryServerFlowImpl.kt]: No - I don't think we want to put these in the exception. If someone tries to double spend someone else's state, we don't want to give them any further information.

**Review by blsemo — CHANGES_REQUESTED** [review]

**Review by blsemo — APPROVED** [review]
LGTM

**Review by lankydan — APPROVED** [review]

**Comment by jennyang-r3** [comment]
build e2e

**Pull request merged**